### PR TITLE
added if checks on CryptoKit and SwiftUI in order to compile on Linux

### PR DESCRIPTION
**Pull request by Cartisim**

### CODE REVIEW

**Comment by Cartisim** [comment]
Please put on hold this PR

**Comment by Cartisim** [comment]
Please resume PR request. I have added URLSession support for Linux in VaporTransport

File: Sources/MessagingHelpers/VaporTransport.swift [hunk 1]
```diff
@@ -129,10 +144,28 @@ extension URLSession {
         let data = try BSONEncoder().encode(body).makeData()
         
         if data.count > maxBodySize {
+#if canImport(FoundationNetworking)
+            guard let res = URLResponse(coder: NSCoder()) else { throw  VaporTransportError.urlResponseNil }
```
Comment (Joannis) [inline Sources/MessagingHelpers/VaporTransport.swift:148]: I'm not a fan of trying to use NSCoder.

**Review by Joannis — COMMENTED** [review]
Looks good, just NSCoder should be gotten rid of

**Comment by Cartisim** [comment]
ok thanks, I am on it

File: Sources/MessagingHelpers/VaporTransport.swift (hunk 1 above)
Comment (Cartisim) [reply Sources/MessagingHelpers/VaporTransport.swift:148]: I think we can initialize URLResponse differently, but I don't think that in FoundationNetworking it comes with an empty initializer, like in Foundation. https://github.com/apple/swift-corelibs-foundation/blob/eec4b26deee34edb7664ddd9c1222492a399d122/Sources/FoundationNetworking/URLResponse.swift

**Pull request closed without merging**